cmd/lakectl/cmd/plugin_list.go

```diff
@@ -0,0 +1,143 @@
+	Name    string
+	Path    string
+	Warning string
+	Verbose bool
+}
+
+var pluginListCmd = &cobra.Command{
+	Use:   "list",
+	Short: "List available lakectl plugins",
+	Long:  `Scans the PATH for executables named "lakectl-*" and lists the detected plugins.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		verbose, _ := cmd.Flags().GetBool("verbose")
+		envPathDirs := filepath.SplitList(os.Getenv("PATH"))
+
+		// Filter out duplicates from envPathDirs
+		var pathDirs []string
+		seen := make(map[string]struct{})
+		for _, dir := range envPathDirs {
+			if dir == "" {
+				// Skip empty entries in PATH
+				continue
+			}
+			if _, ok := seen[dir]; !ok {
+				seen[dir] = struct{}{}
+				pathDirs = append(pathDirs, dir)
+			}
+		}
+
+		// Iterate through pathDirs to build the final list with correct overshadowing
+		pluginsOutput := make([]pluginInfo, 0)
+		foundEffectivePlugins := make(map[string]string) // plugin name -> path of chosen executable
+
+		for _, dir := range pathDirs {
+			files, err := os.ReadDir(dir)
+			if err != nil {
+				// Silently ignore errors from non-existent or unreadable directories in PATH
+				continue
+			}
+
+			for _, file := range files {
+				fileName := file.Name()
+				if !strings.HasPrefix(fileName, "lakectl-") {
+					continue
+				}
+				pluginName := strings.TrimPrefix(fileName, "lakectl-")
+				if pluginName == "" {
+					continue
+				}
+
+				fullPath := filepath.Join(dir, fileName)
+				absPath, err := filepath.Abs(fullPath)
+				if err != nil {
+					// Silently ignore errors from file paths that cannot be resolved
+					continue
+				}
+
+				fileInfo, statErr := file.Info()
+				if statErr != nil {
+					pluginsOutput = append(pluginsOutput, pluginInfo{Name: pluginName, Path: fullPath, Warning: fmt.Sprintf("error getting file info: %v", statErr), Verbose: verbose})
+					continue
+				}
+				if fileInfo.IsDir() {
+					continue
+				}
+
+				var isExec bool
+				if runtime.GOOS == "windows" {
+					// On Windows, check if the file is executable by checking its extension
+					ext := strings.ToLower(filepath.Ext(fileName))
+					isExec = ext == ".exe" || ext == ".com" || ext == ".bat" || ext == ".cmd"
+				} else {
+					// On Unix-like systems, check if the file is executable by user, group, or others
+					isExec = fileInfo.Mode().Perm()&0111 != 0 //nolint:mnd
+				}
+
+				currentPI := pluginInfo{Name: pluginName, Path: absPath, Verbose: verbose}
+				if !isExec {
+					currentPI.Warning = "identified as a lakectl plugin, but it is not executable"
+				}
+
+				if existingPath, ok := foundEffectivePlugins[pluginName]; ok {
+					// Already found an effective plugin for this name. This one is overshadowed.
+					if currentPI.Warning != "" { // It has its own issue (e.g. not executable)
+						currentPI.Warning = fmt.Sprintf("%s (also overshadowed by %s)", currentPI.Warning, existingPath)
+					} else {
+						currentPI.Warning = fmt.Sprintf("overshadowed by %s", existingPath)
```

Comment:
Not a fan. I mean, PATH in exec*p doesn't do it, why should we?

Reply:
Agree, I'll remove the warning from the output and match the first plugin and print them out.